nixos/modules/services/misc/matrix-synapse.nix

```diff
@@ -669,30 +669,43 @@ in {
       gid = config.ids.gids.matrix-synapse;
     };
 
-    services.postgresql = mkIf (usePostgresql && cfg.create_local_database) {
-      enable = mkDefault true;
-      ensureDatabases = [ cfg.database_name ];
-      ensureUsers = [{
-        name = cfg.database_user;
-        ensurePermissions = { "DATABASE \"${cfg.database_name}\"" = "ALL PRIVILEGES"; };
-      }];
-    };
+    services.postgresql.enable = mkIf usePostgresql (mkDefault true);
 
     systemd.services.matrix-synapse = {
       description = "Synapse Matrix homeserver";
-      after = [ "network.target" ] ++ lib.optional config.services.postgresql.enable "postgresql.service" ;
+      after = [ "network.target" "postgresql.service" ];
       wantedBy = [ "multi-user.target" ];
       preStart = ''
         ${cfg.package}/bin/homeserver \
           --config-path ${configFile} \
           --keys-directory ${cfg.dataDir} \
           --generate-keys
+      '' + optionalString (usePostgresql && cfg.create_local_database) ''
+        if ! test -e "${cfg.dataDir}/db-created"; then
```

Comment:
This will cause issues with setups that were created on nixos-unstable while ensureDatabases were used as in those cases a database was created with touching `${cfg.dataDir}/db-created`.

Any ideas atm how we can work around this?

Reply:
@kampka Suggested to use an idempotent SQL statement. This could determine whether the database exists and create it along with the user and role. After the execution, `db-created` will be created, either because the script determined that nothing is to do and exited or because the script created the resources.

sth like
```
SELECT COUNT(*) FROM pg_database WHERE datname="matrix-synapse";
```

Reply:
As mentioned in https://github.com/NixOS/nixpkgs/pull/80447#discussion_r392385305, the db setup will be dropped, so this doesn't need to be discussed anymore.